src/archive.rs

```diff
@@ -36,4 +38,129 @@ pub struct Header {
     pub terminator: [u8; 2],
 }
 
+/// The fixed length header at the start of a big archive.
+#[derive(Debug, Clone, Copy)]
+#[repr(C)]
+pub struct BigFixedLengthHeader {
+    /// Offset to member table.
+    pub memoff: [u8; 20],
+    /// Offset to global symbol table.
+    pub gstoff: [u8; 20],
+    /// Offset global symbol table for 64-bit objects.
+    pub gst64off: [u8; 20],
+    /// Offset to first archive member.
+    pub fstmoff: [u8; 20],
+    /// Offset to last archive member.
+    pub lstmoff: [u8; 20],
+    /// Offset to last archive member.
+    pub freeoff: [u8; 20],
+}
+
+/// The header at the start of a big archive member.
+#[derive(Debug, Clone, Copy)]
+#[repr(C)]
+pub struct BigHeader {
+    /// File size in decimal.
+    pub size: [u8; 20],
+    /// Next member offset in decimal.
+    pub nxtmem: [u8; 20],
+    /// Previous member offset in decimal.
+    pub prvmem: [u8; 20],
+    /// File modification timestamp in decimal.
+    pub date: [u8; 12],
+    /// User ID in decimal.
+    pub uid: [u8; 12],
+    /// Group ID in decimal.
+    pub gid: [u8; 12],
+    /// File mode in octal.
+    pub mode: [u8; 12],
+    /// Name length in decimal.
+    pub namelen: [u8; 4],
+    // Begining of name, or TERMINATOR.
+    // We deal with it manually.
+    // pub name: [u8; 2]
+}
+
+/// Trait for various header style
+pub trait GenericHeader {
+    /// Name as stored in header.
+    fn name(&self) -> &[u8];
+    /// File modification timestamp in decimal.
+    fn date(&self) -> &[u8];
+    /// User ID in decimal.
+    fn uid(&self) -> &[u8];
+    /// Group ID in decimal.
+    fn gid(&self) -> &[u8];
+    /// File mode in octal.
+    fn mode(&self) -> &[u8];
+    /// File size in decimal.
+    fn size(&self) -> &[u8];
+}
+
+impl core::fmt::Debug for dyn GenericHeader {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        write!(f,
+               "Header name: {:?}, date: {:?}, uid: {:?}, gid: {:?},
+               mode: {:?}, size: {:?}.",
+               self.name(), self.date(), self.uid(), self.gid(),
+               self.mode(), self.size())
```

Comment:
I think it would nice if all fields would be interpreted as strings if they are valid UTF-8. Also preferably this would show all fields, not just the common fields.

Reply:
Indeed. As headers derive from Debug, it is possible to write self itself if I am right.